### DISCUSSION TITLE
Restoring `previousRun`

### DISCUSSION THREAD

**Discussion by kinsi55** [Hexagon/croner#207] [Ideas]
In my app I have a couple of tasks which run sparingly (Monthly) but should not be missed. My fear is now that My app may or may not be down at the scheduled runtime - Which is fine - however when it is back up again later the missed task will not be re-run.

The way I see it, adding a way to **set** `previousRun` shouldnt be too complex and essentially fix my problem as I could just store it whenever the task is executed and restore it on app start

**Comment by Hexagon** [comment]
It is possible to get next run time based on previous run time by `job.nextRun(previousRun)`. If the result is before now you have a missed run.

https://croner.56k.guru/usage/basics.html

**Reply by kinsi55** [reply]
I see, that works as a solution for me thank you! Yet I think it could be neat if one could just pass the previousRun time in the options and if done so, Croner handles executing the target by itself assuming its too far in the past
